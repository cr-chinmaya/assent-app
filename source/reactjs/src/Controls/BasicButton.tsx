import * as React from 'react';
import { DefaultButton, PrimaryButton } from '@fluentui/react';

const BasicButton = ({
    text = '',
    title = '',
    onClick = () => {},
    primary = false,
    componentRef = null,
    disabled = false,
    className = '',
    styles = {},
    aiEventName = ''
}: {
    text: string;
    title: string;
    onClick: any;
    primary?: boolean;
    componentRef?: any;
    disabled?: boolean;
    className?: string;
    styles?: object;
    aiEventName?: any
}): any => {
    return primary ? (
        <PrimaryButton
            text={text}
            title={title}
            onClick={onClick}
            componentRef={componentRef}
            disabled={disabled}
            className={className}
            styles={styles}
        />
    ) : (
        <DefaultButton
            text={text}
            title={title}
            onClick={onClick}
            componentRef={componentRef}
            disabled={disabled}
            className={className}
            styles={styles}
        />
    );
};

export default BasicButton;
