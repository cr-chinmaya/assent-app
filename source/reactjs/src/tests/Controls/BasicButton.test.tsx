import * as React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import BasicButton from '../../Controls/BasicButton';

describe('BasicButton', () => {
    it('should BasicButton renders correctly', () => {
        const wrapper = render(<BasicButton />);
        const button = wrapper.container.querySelector('button');
        expect(wrapper).toMatchSnapshot();
        expect(button).toBeTruthy;
    });

    it('should primary Button renders correctly', () => {
        const onClick = () => {};
        const wrapper = render(
            <BasicButton
                text={'Test'}
                title={'Test button'}
                primary={true}
                onClick={onClick}
                componentRef={null}
                disabled={false}
            />
        );
        expect(wrapper.getByTitle('Test button')).toBeTruthy;
    });

    it('Verify BasicButton click event', () => {
        const text = 'test';
        const { getByText } = render(<BasicButton text={text} title={'Testbtn'} primary={true} />);
        const button = getByText(text).closest('button');
        fireEvent.click(button);
    });

    it('should BasicButton click once', () => {
        const handleClick = jest.fn();
        const button = render(<BasicButton title={'Testbtn'} onClick={handleClick} primary={true} />);
        fireEvent.click(button.getByTitle('Testbtn'));
        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
