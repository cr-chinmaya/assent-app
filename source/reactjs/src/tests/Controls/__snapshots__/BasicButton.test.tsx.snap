// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicButton should BasicButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="ms-Button ms-Button--default root-40"
        data-is-focusable="true"
        title=""
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-41"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-label label-44"
            id="id__0"
          />
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ms-Button ms-Button--default root-40"
      data-is-focusable="true"
      title=""
      type="button"
    >
      <span
        class="ms-Button-flexContainer flexContainer-41"
        data-automationid="splitbuttonprimary"
      >
        <span
          class="ms-Button-label label-44"
          id="id__0"
        />
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
